app/actions/remote/thread.ts

```diff
@@ -194,3 +199,152 @@ export const updateThreadFollowing = async (serverUrl: string, teamId: string, t
         return {error};
     }
 };
+
+enum Direction {
+    Up,
+    Down,
+}
+
+async function fetchBatchThreads(
+    serverUrl: string,
+    teamId: string,
+    options: FetchThreadsOptions,
+    pages?: number,
+): Promise<{error: unknown; data?: Thread[]}> {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+
+    if (!operator) {
+        return {error: `${serverUrl} database not found`};
+    }
+
+    let client: Client;
+    try {
+        client = NetworkManager.getClient(serverUrl);
+    } catch (error) {
+        return {error};
+    }
+
+    // if we start from the begging of time (since = 0) we need to fetch threads from newest to oldest (Direction.Down)
+    // if there is another point in time, we need to fetch threads from oldest to newest (Direction.Up)
+    let direction = Direction.Up;
+    if (options.since === 0) {
+        direction = Direction.Down;
+    }
+
+    const currentUser = await getCurrentUser(operator.database);
+    if (!currentUser) {
+        return {error: 'currentUser not found'};
+    }
+
+    const {config} = await getCommonSystemValues(operator.database);
+    const data: Thread[] = [];
+
+    const fetchThreadsFunc = async (opts: FetchThreadsOptions) => {
+        let page = 0;
+        const {before, after, perPage = General.CRT_CHUNK_SIZE, deleted, unread, since} = opts;
+
+        page += 1;
+        const {threads} = await client.getThreads(currentUser.id, teamId, before, after, perPage, deleted, unread, since, false, config.Version);
+        if (threads.length) {
+            // Mark all fetched threads as following
+            for (const thread of threads) {
+                thread.is_following = true;
+            }
+
+            data.push(...threads);
+
+            if (threads.length === perPage) {
+                const newOptions: FetchThreadsOptions = {perPage, deleted, unread};
+                if (direction === Direction.Down) {
+                    const last = threads[threads.length - 1];
+                    newOptions.before = last.id;
+                } else {
+                    const first = threads[0];
+                    newOptions.after = first.id;
+                }
+                if (pages != null && page < pages) {
+                    fetchThreadsFunc(newOptions);
+                }
+            }
+        }
+    };
+
+    try {
+        await fetchThreadsFunc(options);
+    } catch (error) {
+        if (__DEV__) {
+            throw error;
+        }
+
+        return {error, data};
+    }
+
+    return {error: false, data};
```

Comment:
Is there any need to pass data if there is an error or error if there is data? It makes the types a bit less chaotic, I think, but not sure if really needed. 0/5.

Reply:
No need else than simplifying the type